security/kerberos_pac.ksy

```diff
@@ -0,0 +1,542 @@
+        -orig-id: LastFailedILogon
+        type: filetime
+        doc: A FILETIME structure that contains the user account's msDS-LastFailedInteractiveLogonTime ([MS-ADA2] section 2.357)
+      - id: failed_i_logon_count
+        -orig-id: LastFailedILogon
+        type: u4
+        doc: A 32-bit unsigned integer that contains the user account's msDS-FailedInteractiveLogonCountAtLastSuccessfulLogon ([MS-ADA2] section 2.315)
+      - id: reserved3
+        -orig-id: Reserved3
+        type: u4
+      - id: sid_count
+        -orig-id: SidCount
+        type: u4
+        doc: A 32-bit unsigned integer that contains the total number of SIDs present in the ExtraSids member. If this member is not zero then the D bit MUST be set in the UserFlags member
+      - id: extra_sids_referent
+        -orig-id: ExtraSidsReferent
+        type: u4
+      - id: resource_group_domain_sid_referent
+        -orig-id: ResourceGroupDomainSidReferent
+        type: u4
+      - id: resource_group_count
+        -orig-id: ResourceGroupCount
+        type: u4
+        doc: A 32-bit unsigned integer that contains the number of resource group identifiers stored in ResourceGroupIds
+      - id: resource_group_ids_referent
+        -orig-id: ResourceGroupIdsReferent
+        type: u4
+      - id: effective_name_body
+        type: unicode_string_body
+      - id: full_name_body
+        type: unicode_string_body
+      - id: logon_script_body
+        type: unicode_string_body
+      - id: profile_path_body
+        type: unicode_string_body
+      - id: home_directory_body
+        type: unicode_string_body
+      - id: home_directory_drive_body
+        type: unicode_string_body
+      - id: group_ids
+        -orig-id: GroupIds
+        type: rpc_group_membership
+        if: group_count > 0
+      - id: logon_server_body
+        type: unicode_string_body
+      - id: logon_domain_name_body
+        type: unicode_string_body
+      - id: logon_domain_id
+        type: sid
+      - id: extra_sids
+        type: rpc_extra_sids
+        if: (sid_count > 0) and (user_flags & netlogon_extra_sids_user_flag != 0)
+      - id: resource_group_domain_sid
+        -orig-id: ResourceGroupDomainSid
+        type: sid
+        if: user_flags & netlogon_resource_groups_user_flag != 0
+      - id: resource_group_ids
+        -orig-id: ResourceGroupIds
+        type: rpc_resource_group_ids
+        if: (user_flags & netlogon_resource_groups_user_flag != 0) and (resource_group_count > 0)
+  pac_client_info:
+    doc: "See [MS-PAC] 2.7 PAC_CLIENT_INFO"
+    seq:
+      - id: client_id
+        -orig-id: ClientId
+        type: filetime
+        doc: A FILETIME structure in little-endian format that contains the Kerberos initial ticket-granting ticket (TGT) authentication time
+      - id: name_length
+        -orig-id: NameLength
+        type: u2
+        doc: An unsigned 16-bit integer in little-endian format that specifies the length, in bytes, of the Name field
```

Comment:
1. should be called `len_name` instead per [style guide](https://doc.kaitai.io/ksy_style_guide.html#attr-id);
2. such a `doc` is useless and actually counter-productive - only comment what isn't described by the spec.

   Omit any type information - that is already obvious from the declarative description (`/meta/endian: le`, `type: u2`):

   > ~~An unsigned 16-bit integer in little-endian format that~~ specifies the length, in bytes, of the Name field

   Avoid extra wording as _specifies the_ (...), or worse _this is a field that specifies_ (...) - :

   > ~~specifies the~~ length, in bytes, of the Name field

   Now we're left with the only potential valuable info:

   > length, in bytes, of the Name field

   However, see what's written in the [style guide](https://doc.kaitai.io/ksy_style_guide.html#seq-attr):
   
   > `doc` SHOULD NOT be included if it’s trivial (i.e. if it is a copy of `id`, and there is really nothing more to say about that attribute).

   The attribute will be called `len_name`, which already describes "_length, in bytes, of the Name field_" well enough, so we have nothing left to say about it and thus `doc` should be omitted.

Please apply these remarks to the entire file.